sound/soc/sof/pm.c

```diff
@@ -304,6 +308,10 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	struct snd_sof_dev *sdev = dev_get_drvdata(&priv->pdev_pcm->dev);
 	int ret = 0;
 
+	/* probe doesn't finshed, nothing to do for suspend */
+	if (priv->probe_complete == false)
+		return 0;
```

Comment:
@libinyang this is misleading I think. If sof_probe() failed and sdev is null, we should indicate a failure here instead of a success. returning 0 would falsely indicate that suspend was successful.